### PR TITLE
Add meta tensor support for squeeze

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74376
* #74317
* __->__ #74314
* #74313
* #74375

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D34931336](https://our.internmc.facebook.com/intern/diff/D34931336)

### CODE REVIEW

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34931336).

**Review by Chillee — APPROVED** [review]
Kinda surprised this works, but given it (presumably) does, LGTM!

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34931336).

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -4162,15 +4162,15 @@
   device_check: NoCheck
   device_guard: False
   dispatch:
-    CPU, CUDA: squeeze
+    CPU, CUDA, Meta: squeeze
```
[the remark this answers is not in the document]
Comment (ezyang) [reply aten/src/ATen/native/native_functions.yaml:4165]: let's hurry up and merge those PRs then :)

> They actually were previously, and a PR a month or two ago replaced CompositeExplicitAutograd with CPU, CUDA, regressing the coverage for the meta backend (too bad that the test suite didn't catch it :(

I have a new plan for testing based on OpInfos, hopefully will catch this

[the remark this answers is not in the document]
Comment (bdhirsh) [reply aten/src/ATen/native/native_functions.yaml:4165]: nice - landing here (hopefully today): D35016023

**Pull request closed without merging**